src/query/plan/preprocess.hpp

```diff
@@ -153,19 +153,20 @@ struct Expansion {
   ExpansionGroupId expansion_group_id = ExpansionGroupId();
 };
 
+struct PatternComprehensionMatching;
 struct FilterMatching;
 
 enum class PatternFilterType { EXISTS };
 
-/// Collects matchings from filters that include patterns
-class PatternFilterVisitor : public ExpressionVisitor<void> {
+/// Collects matchings that include patterns
+class PatternVisitor : public ExpressionVisitor<void> {
```

Comment:
Why does `PatternFilter` now include `PatternComprehension`?

Reply:
I decided to not have separate class for PatternComprehensionVisitor since it just collects filters and now pattern comprehensions.
If you think that having separate visitor classes will be better, I can add it.

Reply:
I don't see an issue, but maybe @Josipmrden does. So I will let him check this also

Reply:
I think it's fine, as long as there's not too much logic happening. Good to go.